File: server-java-play-framework/app/controllers/WorkflowApiController.java
package controllers;

import apimodels.SirqulResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-11T19:25:37.664944661Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class WorkflowApiController extends Controller {
    private final WorkflowApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private WorkflowApiController(Config configuration, WorkflowApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result runWorkflow(Http.Request request) throws Exception {
        String valueaccountId = request.getQueryString("accountId");
        Long accountId;
        if (valueaccountId != null) {
            accountId = Long.parseLong(valueaccountId);
        } else {
            throw new IllegalArgumentException("'accountId' parameter is required");
        }
        String valueworkflowId = request.getQueryString("workflowId");
        Long workflowId;
        if (valueworkflowId != null) {
            workflowId = Long.parseLong(valueworkflowId);
        } else {
            throw new IllegalArgumentException("'workflowId' parameter is required");
        }
        String valueskuId = request.getQueryString("skuId");
        Long skuId;
        if (valueskuId != null) {
            skuId = Long.parseLong(valueskuId);
        } else {
            skuId = null;
        }
        String valueversionCode = request.getQueryString("versionCode");
        Integer versionCode;
        if (valueversionCode != null) {
            versionCode = Integer.parseInt(valueversionCode);
        } else {
            versionCode = null;
        }
        String valueparameters = request.getQueryString("parameters");
        String parameters;
        if (valueparameters != null) {
            parameters = valueparameters;
        } else {
            parameters = null;
        }
        return imp.runWorkflowHttp(request, accountId, workflowId, skuId, versionCode, parameters);
    }

}

File: server-scala-play-server/app/api/WorkflowApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-11T19:31:43.310890579Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class WorkflowApiController @Inject()(cc: ControllerComponents, api: WorkflowApi) extends AbstractController(cc) {
  /**
    * POST /api/3.18/workflow/run?accountId=[value]&workflowId=[value]&skuId=[value]&versionCode=[value]&parameters=[value]
    */
  def runWorkflow(): Action[AnyContent] = Action { request =>
    def executeApi(): SirqulResponse = {
      val accountId = request.getQueryString("accountId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("accountId", "query string")
        }
        
      val workflowId = request.getQueryString("workflowId")
        .map(value => value.toLong)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("workflowId", "query string")
        }
        
      val skuId = request.getQueryString("skuId")
        .map(value => value.toLong)
        
      val versionCode = request.getQueryString("versionCode")
        .map(value => value.toInt)
        
      val parameters = request.getQueryString("parameters")
        
      api.runWorkflow(accountId, workflowId, skuId, versionCode, parameters)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}
